src/OrchardCore/Orchard.Localization.Core/PoFilesTranslationsProvider.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Options;
+using Orchard.Environment.Extensions;
+using Orchard.Environment.Shell;
+using Orchard.Localization.Abstractions;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Orchard.Localization.Core
+{
+    public class PoFilesTranslationsProvider : ITranslationProvider
+    {
+        private readonly IExtensionManager _extensionsManager;
+        private readonly PoParser _parser;
+        private readonly string _root;
+        private readonly string _rootContainer;
+        private readonly string _shellContainer;
+        private readonly string _shellName;
+
+        public PoFilesTranslationsProvider(
+            IExtensionManager extensionsManager,
+            IHostingEnvironment hostingEnvironment,
+            IOptions<ShellOptions> shellOptions,
+            ShellSettings shellSettings)
+        {
+            _extensionsManager = extensionsManager;
+            _parser = new PoParser();
+
+            _root = hostingEnvironment.ContentRootPath;
+            _rootContainer = shellOptions.Value.ShellsRootContainerName;
+            _shellContainer = shellOptions.Value.ShellsContainerName;
+            _shellName = shellSettings.Name;
+        }
+
+        public void LoadTranslationsToDictionary(string cultureName, CultureDictionary dictionary)
+        {
+            var corePath = Path.Combine(_root, "Core", "App_Data", "Localization", cultureName, "orchard.core.po");
+            LoadFileToDictionary(corePath, dictionary);
+
+            foreach (var extension in _extensionsManager.GetExtensions())
+            {
+                string filename = null;
+                switch (extension.Manifest.Type.ToLowerInvariant())
+                {
+                    case "module": filename = "orchard.module.po"; break;
+                    case "theme": filename = "orchard.theme.po"; break;
+                    default: continue;
+                }
+
+                var extensionPath = Path.Combine(_root, extension.SubPath, "App_Data", "Localization", cultureName, filename);
```

Comment:
"Localization" is no longer hardcoded, but it is loaded from `LocalizationOptions`

`LanguageViewLocationExpander` provides alternative paths for a single view, but we need to get all paths where different .po files are located, so I can't see any way, how we can reuse it. I have implemented `ILocalizationFileLocationProvider` instead. This class returns all locations, where a translation provider should look for .po files.

Still not sure whether I should load "App_Data" and "Localization" parts of path for modules from the configuration. If the path is configurable, module authors wouldn't be able to bundle translations into modules and users will have to copy .po files to correct location manually.